Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -532,36 +571,222 @@ def preprocess_url(url)
       url
     end
 
-    # livecheck should fetch a URL using brewed curl if the formula/cask
+    # livecheck should fetch a URL using brewed curl if the formula/resource/cask
     # contains a `stable`/`url` or `head` URL `using: :homebrew_curl` that
     # shares the same root domain.
-    sig { params(formula_or_cask: T.any(Formula, Cask::Cask), url: String).returns(T::Boolean) }
-    def use_homebrew_curl?(formula_or_cask, url)
+    sig { params(package_or_resource: T.any(Formula, Cask::Cask), url: String).returns(T::Boolean) }
+    def use_homebrew_curl?(package_or_resource, url)
       url_root_domain = Addressable::URI.parse(url)&.domain
       return false if url_root_domain.blank?
 
       # Collect root domains of URLs with `using: :homebrew_curl`
       homebrew_curl_root_domains = []
-      case formula_or_cask
+      case package_or_resource
       when Formula
         [:stable, :head].each do |spec_name|
-          next unless (spec = formula_or_cask.send(spec_name))
+          next unless (spec = package_or_resource.send(spec_name))
           next unless spec.using == :homebrew_curl
 
           domain = Addressable::URI.parse(spec.url)&.domain
           homebrew_curl_root_domains << domain if domain.present?
         end
       when Cask::Cask
-        return false unless formula_or_cask.url.using == :homebrew_curl
+        return false unless package_or_resource.url.using == :homebrew_curl
 
-        domain = Addressable::URI.parse(formula_or_cask.url.to_s)&.domain
+        domain = Addressable::URI.parse(package_or_resource.url.to_s)&.domain
         homebrew_curl_root_domains << domain if domain.present?
       end
 
       homebrew_curl_root_domains.include?(url_root_domain)
     end
 
-    # Identifies the latest version of the formula and returns a Hash containing
+    # Identifies the latest version of the resource in a given Formulae and returns a Hash containing
+    # the version information for a resource. Returns a nil value if a latest version
+    # couldn't be found for a given resource.
+    sig {
+      params(
+        resource:  Resource,
+        json:      T::Boolean,
+        full_name: T::Boolean,
+        verbose:   T::Boolean,
+        debug:     T::Boolean,
+      ).returns(T.nilable(Hash))
+    }
+    def resource_version(
+      resource,
+      json: false,
+      full_name: false,
+      verbose: false,
+      debug: false
+    )
+      has_livecheckable = resource.livecheckable?
+
+      if debug
+        puts "\n\n"
+        puts "Resource:          #{resource.name}"
+        puts "Livecheckable?:    #{has_livecheckable ? "Yes" : "No"}"
```

Comment:
```suggestion
        puts "Resource:         #{resource.name}"
        puts "Livecheckable?:   #{has_livecheckable ? "Yes" : "No"}"
```
For consistency with the formula debug output; the resource debug output wasn't properly aligned.